crates/oxc_language_server/src/main.rs

```diff
@@ -240,34 +240,92 @@ impl LanguageServer for Backend {
-                    title,
-                    kind: Some(CodeActionKind::QUICKFIX),
-                    is_preferred: Some(true),
-                    edit: Some(WorkspaceEdit {
-                        #[expect(clippy::disallowed_types)]
-                        changes: Some(std::collections::HashMap::from([(
-                            uri,
-                            vec![TextEdit {
-                                range: fixed_content.range,
-                                new_text: fixed_content.code,
-                            }],
-                        )])),
-                        ..WorkspaceEdit::default()
+                return Ok(Some(vec![
+                    CodeActionOrCommand::CodeAction(CodeAction {
+                        title: report.diagnostic.message.split(':').next().map_or_else(
+                            || "Fix this problem".into(),
+                            |s| format!("Fix this {s} problem"),
+                        ),
+                        kind: Some(CodeActionKind::QUICKFIX),
+                        is_preferred: Some(true),
+                        edit: Some(WorkspaceEdit {
+                            #[expect(clippy::disallowed_types)]
+                            changes: Some(std::collections::HashMap::from([(
+                                uri.clone(),
+                                vec![TextEdit {
+                                    range: fixed_content.range,
+                                    new_text: fixed_content.code,
+                                }],
+                            )])),
+                            ..WorkspaceEdit::default()
+                        }),
+                        disabled: None,
+                        data: None,
+                        diagnostics: None,
+                        command: None,
+                    }),
+                    CodeActionOrCommand::CodeAction(CodeAction {
+                        title: rule_name.clone().map_or_else(
+                            || "Disable oxlint for this line".into(),
+                            |s| format!("Disable {s} for this line"),
+                        ),
+                        kind: Some(CodeActionKind::QUICKFIX),
+                        is_preferred: Some(false),
+                        edit: Some(WorkspaceEdit {
+                            #[expect(clippy::disallowed_types)]
+                            changes: Some(std::collections::HashMap::from([(
+                                uri.clone(),
+                                vec![TextEdit {
+                                    range: Range {
+                                        start: fixed_content.range.start,
+                                        end: fixed_content.range.start,
+                                    },
+                                    new_text: rule_name.clone().map_or_else(
+                                        || "// eslint-disable-next-line\n".into(),
```

Comment:
Should this be `eslint-disable` or is there an oxlint equivalent available?

Reply:
We do support "oxlint-disable", but I'd be partial to keeping it as eslint-disable just for the sake of compatibility personally